homeassistant/components/tfl/__init__.py

```diff
@@ -0,0 +1,74 @@
+"""The Transport for London integration."""
+
+from __future__ import annotations
+
+import logging
+from urllib.error import HTTPError, URLError
+
+from tflwrapper import stopPoint
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+
+from .const import CONF_API_APP_KEY, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Transport for London from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    app_key = entry.data[CONF_API_APP_KEY]
+    stop_point_api = stopPoint(app_key)
+    try:
+        categories = await hass.async_add_executor_job(
+            stop_point_api.getCategories
+        )  # Check can call endpoint.
+    except HTTPError as exception:
+        # TfL's API returns a 429 if you pass an invalid app_key, but we also check
+        # for other reasonable error codes in case their behaviour changes
+        error_code = exception.code
+        if error_code in (429, 401, 403):
+            raise ConfigEntryAuthFailed(
+                "Authentication failure for app_key=" + app_key
+            ) from exception
+
+        raise ConfigEntryNotReady(
+            "Connection error whilst connecting to TfL"
+        ) from exception
+    except URLError as exception:
```

Comment:
The position of the blank line is a bit strange to my eyes. Maybe do:

```suggestion

        raise ConfigEntryNotReady(
            "Connection error whilst connecting to TfL"
        ) from exception

    except URLError as exception:
```